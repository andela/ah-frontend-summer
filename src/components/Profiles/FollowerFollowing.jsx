import React from 'react';
import { Container } from 'semantic-ui-react';
import PropTypes from 'prop-types';
import User from './User';

const FollowerFollowing = props => {
    const { 
        location: { pathname } = {}, 
        profile: { followers } = {},
        profile: { followings } = {},
        match: { params: { username} } = { params: {} }
    } = props;
    const users = pathname === `/${username}/followers` ? followers : followings;
    let title;
    let noUsers;
    if (pathname === `/${username}/followers`){ 
        title = `${username} Followers`;
        noUsers = `${username} has no followers yet`;
    }

    if (pathname === `/${username}/following`){ 
        title = `${username} Following`;
        noUsers = `${username} is not following anybody yet`;
    }

    return (
        <Container centered>
            <h1>{title}</h1>
            {!users || users.length === 0 ? <h3>{noUsers}</h3> : users.map(user => (
                <User key={user} user={user} />
            ))}
        </Container>
    );
};

FollowerFollowing.propTypes = {
    profile: PropTypes.object.isRequired,
    location: PropTypes.object.isRequired,
    match: PropTypes.object.isRequired
};

export default FollowerFollowing; 
