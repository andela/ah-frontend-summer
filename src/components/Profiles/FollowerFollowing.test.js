import React from 'react';
import { shallow } from 'enzyme';
import FollowerFollowing from './FollowerFollowing';

describe("Follow Following", () => {
    let props = {
        profile: { followers: [], followings: [] },
        location: { pathname: "" },
        match: { params: { username: ""} }
    };
    let followFollowing = shallow(<FollowerFollowing {...props}/>);

    it("renders correctly", () => {
        expect(followFollowing).toMatchSnapshot();
    });

    it("renders correctly with followers", () => {
        followFollowing = shallow(<FollowerFollowing {...props} />)
        expect(followFollowing).toMatchSnapshot();
    });
});
