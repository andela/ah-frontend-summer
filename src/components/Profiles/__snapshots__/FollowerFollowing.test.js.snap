// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Follow Following renders correctly 1`] = `
<Container
  centered={true}
>
  <h1 />
  <h3 />
</Container>
`;

exports[`Follow Following renders correctly with followers 1`] = `
<Container
  centered={true}
>
  <h1 />
  <h3 />
</Container>
`;
